### PR TITLE
Update participants script ...

### PR DESCRIPTION
**Pull request by gkellogg**
…  to maintain the order of users from the API and make APIKEY optional. Also updates the participants to the current list of members.

### CODE REVIEW

**Comment by gkellogg** [comment]
Once merged, PRs on other repositories will fail until their copy of participants.html matches this version.

**Comment by gkellogg** [comment]
@TallTed I called upon ChatGPT to give me some javascript to approximate the aligning of the results you suggested. in https://github.com/w3c/rdf-primer/pull/16#discussion_r1887171035.

**Review by niklasl — APPROVED** [review]
Looks good to me.

File: participants.js [hunk 1]
```diff
@@ -53,16 +50,24 @@ https.get(groupOptions, (res) => {
...
-          const joinedUsers = sortedUsers.join(' and ');
+        // Find the maximum length of the first part (before the space)
+        const maxLength = Math.max(...users.map(user => user.split(" ")[0].length));
+
+        // Right-align the first component and format the strings
+        const alignedUsers = users.map(user => {
+          const [first, ...rest] = user.split(" ");
+          return first.padStart(maxLength, " ") + " " + rest.join(" ");
+        });
+
```
Comment (pchampin) [inline participants.js]:
```suggestion
```
Move this in the `else` branch where it is actually needed.
(see below)

File: participants.js [hunk 2]
```diff
@@ -53,16 +50,24 @@ https.get(groupOptions, (res) => {
...
+        if (users.length === 1) {
+          console.log(`The sole member of the ${groupName} Group was ${users[0]}.`);
+        } else if (users.length === 2) {
+          const joinedUsers = users.join(' and ');
           console.log(`Members of the ${groupName} Group included ${joinedUsers}.`);
         } else {
-          const joinedUsers = sortedUsers.slice(0, -1).join(', ');
-          console.log(`Members of the ${groupName} Group included ${joinedUsers}, and ${sortedUsers[sortedUsers.length - 1]}.`);
+          const joinedUsers = alignedUsers.slice(0, -1).join(",\n");
```
Comment (pchampin) [inline participants.js:68]: (see above)
```suggestion
          // Find the maximum length of the first part (before the space)
          const maxLength = Math.max(...users.map(user => user.split(" ")[0].length));

          // Right-align the first component and format the strings
          const alignedUsers = users.map(user => {
            const [first, ...rest] = user.split(" ");
            return first.padStart(maxLength, " ") + " " + rest.join(" ");
          });
          const joinedUsers = alignedUsers.slice(0, -1).join(",\n");
```

**Review by pchampin — CHANGES_REQUESTED** [review]
Beyond the proposed change in the code, one question:
this change is only visible in the source HTML, right?
The rendering will not be affected (and to be clear, I don't think it should).

**Comment by gkellogg** [comment]
> Beyond the proposed change in the code, one question: this change is only visible in the source HTML, right? The rendering will not be affected (and to be clear, I don't think it should).

Other than the order of participants, it should render the same.

**Review by pchampin — APPROVED** [review]

**Comment by TallTed** [comment]
- I noted this concern elsewhere, that some people may still be sorted incorrectly (e.g., `Jeffrey Phillips Freeman`, `Thomas Pellissier Tanon`). 

- I note now that `ROSSI` should be changed to `Rossi`, in whatever the ultimate source is.

- I also note that this script grabs "today's" WG membership list, which I presume does not include members who have *left* the group after making contributions. This is a parallel concern to "run the script as late as possible, to include recent joins".

**Comment by gkellogg** [comment]
> * I noted this concern elsewhere, that some people may still be sorted incorrectly (e.g., `Jeffrey Phillips Freeman`, `Thomas Pellissier Tanon`).

Sorting comes from the W3C API, so any changes would need to go into their system.

> * I note now that `ROSSI` should be changed to `Rossi`, in whatever the ultimate source is.

Again, that's the way it is in the user record, so that would be the place to change it.

> * I also note that this script grabs "today's" WG membership list, which I presume does not include members who have _left_ the group after making contributions. This is a parallel concern to "run the script as late as possible, to include recent joins".

[The API](https://api.w3.org/doc) has an option to include former participants; we can update the script to set this flag in the future. As you can see, with this workflow, there are a lot of places to do updates, which must be consistent with this repository. I've run the script and done PRs against the various repositories by hand; we can redo it on demand, and should certainly do so before CR. Otherwise, when someone asks for them to be updated.

**Comment by gkellogg** [comment]
FYI, adding the `former=true` query option adds the following former participants:

           Dan Brickley,
        Sarven Capadisli,
         David Chaves-Fraga,
          Alan Snyder, and
        Stuart Sutton.

**Comment by TallTed** [comment]
> Sorting comes from the W3C API, so any changes would need to go into their system.

I have no argument with this, but I am not well equipped to log an appropriate issue. I hope you are better equipped. I am nonetheless happy to contribute to discussion of the matter, so please feel free to tag me in whatever related issue(s) you may log.

> the [W3C] user record [...] would be the place to change it

Perhaps. I think that the info in "the [W3C] user record" gets put to a variety of uses, and it is *possible* that `ROSSI` is the appropriate capitalization for one or more of these. Still, it is certainly *not* the appropriate capitalization for this and any other contributor lists, where `Rossi` would be appropriate, unless all other family names are changed to all-caps.

> with this workflow, there are a lot of places to do updates, which must be consistent with this repository

Can ReSpec's [data-include](https://respec.org/docs/#data-include) attribute not be put to use for these _[External Includes](https://respec.org/docs/#external-includes)?_

**Comment by gkellogg** [comment]
>>Sorting comes from the W3C API, so any changes would need to go into their system.

>I have no argument with this, but I am not well equipped to log an appropriate issue. I hope you are better equipped. I am nonetheless happy to contribute to discussion of the matter, so please feel free to tag me in whatever related issue(s) you may log.

I have found no issues with the way the names are sorted, so nothing to file. As acknowledgements are typically presented as a comma-separated list of names, the API accomplishes that as is.

>> the [W3C] user record [...] would be the place to change it

> Perhaps. I think that the info in "the [W3C] user record" gets put to a variety of uses, and it is possible that ROSSI is the appropriate capitalization for one or more of these. Still, it is certainly not the appropriate capitalization for this and any other contributor lists, where Rossi would be appropriate, unless all other family names are changed to all-caps.

It seems to me that by setting up the user record this way, Jean-Yves is setting his preference; if not, the appropriate place to fix it would be to update his user record. Failing this, a team member, such as @pchampin, can probably get it updated.

> Can ReSpec's [data-include](https://respec.org/docs/#data-include) attribute not be put to use for these _[External Includes](https://respec.org/docs/#external-includes)?_

Unfortunately, not; we tried this for JSON-LD some time ago, and it doesn't load this way, likely due to CORS issues. Git sub-modules would be an alternative, but these tend to be difficult to work with, so we've gone towards manual updates. In any case, it would require updating the commit number of that sub-module when changed, which ends up being another local commit.

**Comment by TallTed** [comment]
Maybe @JyRossi could chime in, too?

**Pull request merged**